src/components/AssetManagementPanel/AssetManagementForm/AssetManagementForm.tsx

```diff
@@ -39,32 +39,37 @@ export const AssetManagementForm = ({
   const isBeneficiary = account === beneficiary;
   const canSurrender = isBeneficiary && isHolder;
 
+  const onBack = (isPendingConfirmation: boolean) => {
+    if (isPendingConfirmation) return;
+    onSetFormAction(AssetManagementActions.None);
+  };
+
   switch (formAction) {
     case AssetManagementActions.Surrender:
       return (
         <SurrenderForm
           formAction={formAction}
-          onSetFormAction={onSetFormAction}
           tokenId={tokenId}
           tokenRegistryAddress={tokenRegistryAddress}
           beneficiary={beneficiary}
           holder={holder}
           handleSurrender={onSurrender}
           surrenderingState={surrenderingState}
+          onBack={onBack}
```

Comment:
```suggestion
          onBack={()=>onSetFormAction(AssetManagementActions.None)}
```

or you can have define it once and use a few times later

```
const handleBack = () => onSetFormAction(AssetManagementActions.None);

...

onBack={handleBack}
```

The reason is that the surrenderingState is already passed into the form, the parent provides no additional value by having the function do the additional check. In fact, now the components become more tightly coupled because the child need to know that there is one more boolean to pass in. If that's the case, then it's better for the child to do it.

Alternatively if this is to be handled at the parent state then, it will be for the parent to calculate when is it disabled instead of delegating back to the child (so remove the boolean). So you really need to choose if it is to be handled by the parent or the child. Either way, the onBack should be a function without params.

Reply:
Hmm... ok understood, updated the component.